cpmpy/solvers/choco.py

```diff
@@ -0,0 +1,585 @@
+                    elif not isinstance(rhs, _NumVarImpl):
+                        raise Exception(
+                            f"Choco does not accept {rhs} with type {type(rhs)} as rhs of expression {lhs.name}")
+                    arr, val = self.solver_vars(lhs)
+                    return self.chc_model.count(val, arr, chcrhs)
+                elif lhs.name == 'mul':
+                    return self.chc_model.times(self.solver_vars(lhs.args[0]), self.solver_vars(lhs.args[1]),
+                                                chcrhs)
+                elif lhs.name == 'div':
+                    # Choco needs divisor to be a variable
+                    if isinstance(lhs.args[1], int):
+                        divisor = self.to_var(lhs.args[1])  # convert to "variable"
+                    elif isinstance(lhs.args[1], _NumVarImpl):
+                        divisor = self.solver_var(lhs.args[1])  # use variable
+                    else:
+                        raise Exception(f"Choco does not accept {rhs} with type {type(rhs)} as rhs of expression {lhs.name}")
+                    # Choco needs result to be a variable
+                    if isinstance(rhs, int):
+                        result = self.to_var(rhs)  # convert to "variable"
+                    elif isinstance(rhs, _NumVarImpl):
+                        result = self.solver_var(rhs)  # use variable
+                    else:
+                        raise Exception(f"Choco does not accept {rhs} with type {type(rhs)} as rhs of expression {lhs.name}")
+                    return self.chc_model.div(self.solver_var(lhs.args[0]), divisor, result)
+                elif lhs.name == 'element':
+                    if isinstance(rhs, int):
+                        result = self.to_var(rhs)  # convert to "variable"
+                    elif isinstance(rhs, _NumVarImpl):
+                        result = self.solver_var(rhs)  # use variable
+                    else:
+                        raise Exception(f"Choco does not accept {rhs} with type {type(rhs)} as rhs of expression {lhs.name}")
+                    return self.chc_model.element(result, self.solver_vars(lhs.args[0]),
+                                                  self.solver_var(lhs.args[1]))
+                elif lhs.name == 'mod':
+                    divisor = lhs.args[1]
+                    if not isinstance(rhs, _NumVarImpl): # if divisor is variable then result must be variable too
+                        rhs = self.to_var(rhs)  # convert to "variable"
+                    else:
+                        rhs = self.solver_vars(rhs)  # get choco variable
+                    return self.chc_model.mod(self.solver_var(lhs.args[0]), self.solver_var(divisor), rhs)
+                elif lhs.name == 'pow':
+                    if isinstance(rhs, int):     # Choco does not accept an int in rhs
+                        chcrhs = self.to_var(rhs)  # convert to "variable"
```

Comment:
same as for div, 2 mod IV also gives an error, so all arguments need to be converted to 'variable'